### PR TITLE
improve deploying contracts tutorial

### LINKED ISSUE #760: Tutorial page needs more detail: Deploying Contracts
This page is a bit confusing. It starts off spelling everything out, like a tutorial, but then starts leaving gaps so beginners are getting lost.

eg 
>Am a little lost trying to deploy my smart contract using Web3js-quorum following this tutorial: https://besu.hyperledger.org/en/stable/Tutorials/Contracts/Deploying-Contracts/ So I create a file named compile.js, save the first code block on that page then run npm compile.js . Where do I save the second code block starting with const web3 = new Web3(host); ? I cannot see the connection between these 2 'files' and web3js-quorum ?

Check the issue type in the following list (insert X instead of space between [ ]): -->

- [x] Missing content
- [ ] Outdated content
- [ ] Wrong content
- [x] Confusing or misleading content
- [ ] Other

## The broken page

https://besu.hyperledger.org/en/latest/Tutorials/Contracts/Deploying-Contracts/

## Change suggestion

Suggest someone work through the tutorial and add missing info where necessary

### PR DESCRIPTION
**Pull request by alexandratran**
## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [ ] you fixed all the issues raised by the tests, if any.
- [ ] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change

Test and improve Deploying Contracts tutorial to add more detail based on user feedback.

## Issue fixed

fixes #760 

### For content changes

- [ ] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

### CODE REVIEW

File: docs/Tutorials/Contracts/Deploying-Contracts.md [hunk 1]
```diff
@@ -134,16 +132,21 @@ The Developer Quickstart provides an [example of a public transaction script](ht
...
 ## Using `eth_sendTransaction`
 
-Use [`eth_sendTransaction`](https://eth.wiki/json-rpc/API) as an alternative to `eth_sendSignedTransaction`.
-However, Hyperledger Besu does not support the `eth_sendTransaction` API call and keeps account
-management separate for stronger security. Instead, Besu uses
-[EthSigner](https://docs.ethsigner.consensys.net/en/stable/) to make the `eth_sendTransaction` API call.
+You can use [`eth_sendTransaction`](https://eth.wiki/json-rpc/API) as an alternative to `eth_sendSignedTransaction`.
+However, Hyperledger Besu does not support the `eth_sendTransaction` API call and keeps account management separate for
+stronger security.
+Instead, Besu uses [EthSigner](https://docs.ethsigner.consensys.net/en/stable/) to make the `eth_sendTransaction` API call.
```
Comment (macfarla) [inline docs/Tutorials/Contracts/Deploying-Contracts.md:149]: I would say 
You can configure EthSigner with your Besu node to make the eth_sendTransaction call
or something like that - Besu doesn't know anything about EthSigner

File: docs/Tutorials/Contracts/Deploying-Contracts.md [hunk 2]
```diff
@@ -4,11 +4,11 @@ description: deploying smart contracts
...
-networks.
-
 This tutorial shows you how to deploy smart contracts as transactions to a network.
 
+You can use the [Developer Quickstart](../Developer-Quickstart.md) to rapidly generate local blockchain
+networks.
+
 ## Prerequisites
 
 * A [private network](../Examples/Private-Network-Example.md) if deploying a public contract.
```
Comment (macfarla) [inline docs/Tutorials/Contracts/Deploying-Contracts.md:14]: realise you didn't change this - but the "private network" vs "privacy-enabled network" is a bit confusing. have we tried to come up with something better?

actually maybe the "you can use the quickstart" from line 9 would fit in here - this tutorial uses a local blockchain, you can use the quickstart to rapidly generate one... and if deploying a private contract, you need to enable privacy

File: docs/Tutorials/Contracts/Deploying-Contracts.md [hunk 3]
```diff
@@ -28,40 +28,40 @@ Using the
...
+      return {
+        abi: artifact.abi,
+        bytecode: artifact.evm.bytecode.object,
+      };
+    }
+    
+    main().then(() => process.exit(0));
+    ```
 
 Run the compile code to get the smart contract's output JSON:
```
Comment (macfarla) [inline docs/Tutorials/Contracts/Deploying-Contracts.md:66]: Run `compile.js` to get...

File: docs/Tutorials/Contracts/Deploying-Contracts.md [hunk 4]
```diff
@@ -134,16 +132,21 @@ The Developer Quickstart provides an [example of a public transaction script](ht
     so we append that to the binary value).
```
Comment (macfarla) [inline docs/Tutorials/Contracts/Deploying-Contracts.md:132]: in this section I would change
* `from` - address of the EthSigner account. For example `0xfe3b557e8fb62b89f4916b721be55ceb828dbd73`.
to
* `from` - address of the sending account. For example `0xfe3b557e8fb62b89f4916b721be55ceb828dbd73`.

File: docs/Tutorials/Contracts/Deploying-Contracts.md [hunk 5]
```diff
@@ -155,18 +158,6 @@ Pass the following parameters to the
...
-  "gas": "0x76c0",
-  "gasPrice": "0x9184e72a000",
-  "data": "0x608060405234801561001057600080fd5b5060405161014d38038061014d8339818101604052602081101561003357600080fd5b8101908080519060200190929190505050806000819055505060f38061005a6000396000f3fe6080604052348015600f57600080fd5b5060043610603c5760003560e01c80632a1afcd914604157806360fe47b114605d5780636d4ce63c146088575b600080fd5b604760a4565b6040518082815260200191505060405180910390f35b608660048036036020811015607157600080fd5b810190808035906020019092919050505060aa565b005b608e60b4565b6040518082815260200191505060405180910390f35b60005481565b8060008190555050565b6000805490509056fea2646970667358221220e6966e446bd0af8e6af40eb0d8f323dd02f771ba1f11ae05c65d1624ffb3c58264736f6c63430007060033"
-}
-```
-
-The parameters are:
-
 * `to` - address of the receiver. To deploy a contract, set to `null`.
 * `from` - address of the EthSigner account. For example `0x9b790656b9ec0db1936ed84b3bea605873558198`.
```
Comment (macfarla) [inline docs/Tutorials/Contracts/Deploying-Contracts.md:162]: s/EthSigner/sender
it does have to be configured in EthSigner but that would be covered in the EthSigner setup docs

**Review by macfarla — APPROVED** [review]
some comments. nothing major.

**Review by rolandtyler — APPROVED** [review]
lgtm

**Pull request merged**